corehq/apps/users/tests/test_deactivate_mobile_worker_trigger.py

```diff
@@ -115,3 +116,113 @@ def test_users_deactivated(self):
+        )
+
+    def test_creates(self):
+        status = DeactivateMobileWorkerTrigger.update_trigger(
+            'test-trigger', 'user-trigger-002', datetime.date(2022, 2, 1)
+        )
+        self.assertEqual(
+            status, DeactivateMobileWorkerTriggerUpdateMessage.CREATED
+        )
+        trigger = DeactivateMobileWorkerTrigger.objects.get(
+            domain='test-trigger',
+            user_id='user-trigger-002'
+        )
+        self.assertEqual(
+            trigger.deactivate_after,
+            datetime.date(2022, 2, 1)
+        )
+
+    def test_deletes(self):
+        DeactivateMobileWorkerTrigger.objects.create(
+            domain='test-trigger',
+            user_id='user-trigger-003',
+            deactivate_after=datetime.date(2022, 3, 1),
+        )
+        status = DeactivateMobileWorkerTrigger.update_trigger(
+            'test-trigger', 'user-trigger-003', None
+        )
+        self.assertEqual(
+            status, DeactivateMobileWorkerTriggerUpdateMessage.DELETED
+        )
+        self.assertFalse(
+            DeactivateMobileWorkerTrigger.objects.filter(
+                domain='test-trigger',
+                user_id='user-trigger-003'
+            ).exists()
+        )
+
+    def test_no_update_needed(self):
+        existing_trigger = DeactivateMobileWorkerTrigger.objects.create(
+            domain='test-trigger',
+            user_id='user-trigger-001',
+            deactivate_after=datetime.date(2022, 2, 1),
+        )
+        status = DeactivateMobileWorkerTrigger.update_trigger(
+            'test-trigger', 'user-trigger-001', datetime.date(2022, 2, 1),
+        )
+        self.assertIsNone(status)
+        existing_trigger.refresh_from_db()
+        self.assertEqual(
+            existing_trigger.deactivate_after,
+            datetime.date(2022, 2, 1)
+        )
+
+    def test_noop(self):
+        status = DeactivateMobileWorkerTrigger.update_trigger(
+            'test-trigger', 'user-trigger-004', None
+        )
+        self.assertIsNone(status)
+        self.assertFalse(
+            DeactivateMobileWorkerTrigger.objects.filter(
+                domain='test-trigger',
+                user_id='user-trigger-004'
+            )
+        )
+
+    def test_string_value(self):
+        status = DeactivateMobileWorkerTrigger.update_trigger(
+            'test-trigger', 'user-trigger-005', '03-2022'
+        )
+        self.assertEqual(
+            status, DeactivateMobileWorkerTriggerUpdateMessage.CREATED
+        )
+        trigger = DeactivateMobileWorkerTrigger.objects.get(
+            domain='test-trigger',
+            user_id='user-trigger-005'
+        )
+        self.assertEqual(
+            trigger.deactivate_after,
+            datetime.date(2022, 3, 1)
+        )
+
+    def test_bad_value(self):
+        with self.assertRaises(ValueError):
+            DeactivateMobileWorkerTrigger.update_trigger(
+                'test-trigger', 'user-trigger-006', 'foobar'
```

Comment:
Would it be worth adding a test that passes a value of `01-02-2023`? I assume you'd want that to fail.

Reply:
291d4260e05c4a48a3080a2c5b8b028f7a2ffa98